src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/scaleset/AzureScaleSetProvider.java

```diff
@@ -0,0 +1,382 @@
+                           .withFrontendPort(81)
+                           .withProbe(azureTCPProbeName)
+                           .withBackend(azureBackendPoolName)
+                           .attach()
+                           // Define NAT pools for SSH, PNP and PNPS
+                           .defineInboundNatPool(azureNATPoolSSHName)
+                           .withProtocol(TransportProtocol.TCP)
+                           .withFrontend(azureFrontendName)
+                           .withFrontendPortRange(50000, 50099)
+                           .withBackendPort(22) // SSH
+                           .attach()
+
+                           .defineInboundNatPool(azureNATPoolPNPName)
+                           .withProtocol(TransportProtocol.TCP)
+                           .withFrontend(azureFrontendName)
+                           .withFrontendPortRange(30000, 30099)
+                           .withBackendPort(64738) // PNP
+                           .attach()
+                           .defineInboundNatPool(azureNATPoolPNPSName)
+                           .withProtocol(TransportProtocol.TCP)
+                           .withFrontend(azureFrontendName)
+                           .withFrontendPortRange(40000, 40099)
+                           .withBackendPort(64739) // PNPS
+                           .attach()
+                           .create();
+    }
+
+    private VirtualMachineScaleSetSkuTypes selectScaleSetSku(String instanceType) {
+        if (instanceType == null)
+            return VirtualMachineScaleSetSkuTypes.STANDARD_D1_V2;
+        if (instanceType.contains("STANDARD_D1_V2"))
+            return VirtualMachineScaleSetSkuTypes.STANDARD_D1_V2;
+        //TODO: add support for more SKU
+        return VirtualMachineScaleSetSkuTypes.STANDARD_D1_V2;
+    }
+
+    private KnownLinuxVirtualMachineImage selectScaleSetLinuxImage(String imageNameOrId) {
+        if (imageNameOrId == null)
+            return KnownLinuxVirtualMachineImage.UBUNTU_SERVER_16_04_LTS;
+        if (imageNameOrId.contains("Debian"))
+            return KnownLinuxVirtualMachineImage.DEBIAN_8;
+        if (imageNameOrId.contains("CentOS"))
+            return KnownLinuxVirtualMachineImage.CENTOS_7_2;
+        if (imageNameOrId.contains("Ubuntu"))
+            return KnownLinuxVirtualMachineImage.UBUNTU_SERVER_16_04_LTS;
+        return KnownLinuxVirtualMachineImage.UBUNTU_SERVER_16_04_LTS; // default
+    }
+
+    @Override
+    public void deleteInstance(Infrastructure infrastructure, String instanceId) {
+        Azure azureService = azureServiceCache.getService(infrastructure);
+
+        final int azureResourcesUUID = Math.abs(infrastructure.hashCode()); // Make it distinguable and findable (OK since 1 NS <-> 1 Scale Set)
+        final String azureLoadBalancerName = "LB-" + azureResourcesUUID;
+        final String azureVNetName = "VNET-" + azureResourcesUUID;
+        final String azureIPName = "IP-" + azureResourcesUUID;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "azureIPName" which hides the field declared at line 98. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AHiddenFieldCheck)